docs/source/dev/test.rst

```diff
@@ -0,0 +1,92 @@
+.. highlight:: cpp
+
+Writing a unit test
+==================
+
+After implementing a new functionality in Alpaka, it is recommended to test it. Indeed, having no compile time errors
+when building alpaka with any backend doesn't mean that such functionality is well implemented and behaves as expected.
+
+Unit tests are written and integrated with `Catch2 <https://github.com/catchorg/Catch2>`_ and they are standalone executables located in the ``test/unit`` and
+``test/integ`` paths inside the ``build`` folder. They can be built all at once by setting the ``CMake``
+argument ``-DBUILD_TESTING=ON`` and running the command ``cmake --build .`` in the ``build`` folder, and then run with the
+``ctest`` command as described in the :doc:`Installation section </basic/install>`.
+
+In alternative, a target test can be built by adding the flag ``--target <testname>`` to the ``cmake --build`` command and
+then executed as a normal binary file (``./<testname>``) in the corresponding folder mentioned above.
+
+Sample workflow
+-------------------
```

Comment:
```suggestion
Sample workflow
---------------
```
Ok, maybe I messed that up previously. The goal here is to make the line as long as the heading's text above it.